### PR TITLE
Clear input rebinding controls on options exit

### LINKED ISSUE #4032: Clear the input rebinding controls when options menu is closed
as otherwise the buttons just stay in the background forever listening for each input event, which is completely unnecessary drain on performance. A simple method on `InputGroupList` when the options is exited (there's already detection for that for the unsaved settings check) to clear its children should be enough.

### PR DESCRIPTION
**Pull request by d0tplus**
**Brief Description of What This PR Does**

Clears input rebinding controls on options exit. 

**Related Issues**

Closes #4032 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [X] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/engine/input/key_mapping/InputGroupList.cs [hunk 1]
```diff
@@ -178,6 +178,11 @@ public void InitGroupList()
         EnsureNavigationFlowIsCorrect();
     }
 
+    public void ClearGroupList()
+    {
+        this.FreeChildren();
```
Comment (hhyyrylainen) [inline src/engine/input/key_mapping/InputGroupList.cs:183]: Related to my comment on the issue, for example `activeInputGroupList` is something that immediately pops out to me as a potential place where there's probably stored Controls that should be cleared.

Comment (d0tplus) [reply src/engine/input/key_mapping/InputGroupList.cs]: Thank you for the advice! 

I looked over my code, and the error logs in more detail. I think I found and clear references to children that were freed. 

I am now getting the correct behavior, and no longer getting the error from before.

**Comment by hhyyrylainen** [comment]
Other than the CI failure (you removed a BOM mark from a file) this seems to be ready for reviews now.

**Review by hhyyrylainen — APPROVED** [review]
Code change looks good. I also tested with a debugger in the game to confirm that the constant input processing the issue this solves was opened about is properly fixed. 
Good work.

**Pull request merged**